src/js/workshop-comments.js

```diff
@@ -0,0 +1,144 @@
+	const commentDOMElement = document.getElementById("comment");
+	const comment = commentDOMElement.value.trim();
+
+	const submittedCommentMessage = document.querySelector(".post-failure-message");
+
+	disableFields(true);
+
+	// Hide required message
+	// Note: It's necessary to hide "required" indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment
+	document.querySelectorAll(".required").forEach(element => {
+		applyVisibility(element, false);
+	});
+	// Hide comment submission message
+	// Note: It's necessary to hide succesful/failed submitted comment message indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment
+	document.querySelectorAll(".save-response").forEach(element => {
+		applyVisibility(element, false);
+	});
+
+	// Verify input Fields
+	if ((name === "") & (comment === "")) {
+		document.querySelectorAll(".required").forEach(element => {
+			applyVisibility(element, true, "inline");
+		});
+		disableFields(false);
+		return;
+	}
+
+	if (name === "") {
+		applyVisibility(document.querySelector(".required-name"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	if (comment === "") {
+		applyVisibility(document.querySelector(".required-comment"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	const XHR = new XMLHttpRequest();
+
+	const commentForm = document.getElementById("comment-form");
+	// Bind the FormData object and the form element
+	const commentFormData = new FormData( commentForm );
+
+	// submit and add comment to top of comment section of page on successful data submission
+	XHR.addEventListener( "load", function() {
+		if (XHR.status === 200) {
+			setSubmittedComment();
+			// Clear Text Fields
+			commentForm.reset();
+		} else {
+			// Show failed comment submission message
+			submittedCommentMessage.setAttribute("style", "display: block;");
+		}
+		disableFields(false);
+	});
+
+	// Show failure message in case of error
+	XHR.addEventListener( "error", function() {
+		// Show failed comment submission message
+		submittedCommentMessage.setAttribute("style", "display: block;");
+		disableFields(false);
+	} );
+
+	// Set up our request
+	XHR.open( "POST", window.location.origin + "/api/comments" );
+
+	// The data sent is what the user provided in the form
+	const jsonFormData = JSON.stringify(Object.fromEntries(commentFormData));
+	XHR.setRequestHeader("Content-Type", "application/json");
+	XHR.send(jsonFormData);
+});
+
+/*
+ * Hide/show required and success messages
+ * @param {DOMElement} DOMselector - The DOM Element to parse.
+ * @param {Boolean} toDisplay - value is true to show and false to hide
+ * @param {String}[Optional] displayType - CSS display value to apply
+ */
+function applyVisibility(DOMselector, toShow, displayType = "none") {
```

Comment:
It doesn't make sense to set the default `displayType` to `none` as the `displayType` will be applied when showing the DOM content. It would make more sense to set the default value to "block".

Rename "DOMselector" -> "domSelector".